File: 804-unique-morse-code-words/804-unique-morse-code-words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        vector<string> morse = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        set<string> s;
        for(string word : words){
            string morse_code = "";
            for(char ch : word){
                for(char c : morse[ch-'a']){
                    morse_code.push_back(c);
                }
            }
            s.insert(morse_code);
        }
        return s.size();
    }
};

File: 804-unique-morse-code-words/804-unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        s = set();
        morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        for word in words:
            morse_code = []
            for ch in word:
                for c in morse[ord(ch)-ord('a')]:
                    morse_code.append(c);
            s.add(''.join(morse_code));
            
        return len(s);